src/envoy/http/authn/http_filter_factory.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "envoy/registry/registry.h"
+#include "google/protobuf/util/json_util.h"
+#include "src/envoy/http/authn/http_filter.h"
+#include "src/envoy/http/authn/policy.pb.validate.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+class AuthnFilterConfig : public NamedHttpFilterConfigFactory,
+                          public Logger::Loggable<Logger::Id::http> {
+ public:
+  HttpFilterFactoryCb createFilterFactory(const Json::Object& config,
+                                          const std::string&,
+                                          FactoryContext& context) override {
+    istio::authentication::v1alpha1::Policy proto_config;
+
+    ENVOY_LOG(debug, "Called AuthnFilterConfig : {}", __func__);
+
+    MessageUtil::loadFromJson(config.asJsonString(), proto_config);
```

Comment:
Use Utils::ParseJsonMessage then you can simply link to the proto defined in api repo instead of copying them. I believe this function simply has some extra validation, which require to envoy validation.proto. We can ignore all that for now.

Reply:
Done.